onmt/trainer.py

```diff
@@ -112,18 +118,26 @@ def __init__(self, model, train_loss, valid_loss, optim,
         self.average_every = average_every
         self.model_dtype = model_dtype
 
-        assert grad_accum_count > 0
-        if grad_accum_count > 1:
-            assert self.trunc_size == 0, \
-                """To enable accumulated gradients,
-                   you must disable target sequence truncating."""
+        for i in range(len(self.accum_count_l)):
+            assert self.accum_count_l[i] > 0
+            if self.accum_count_l[i] > 1:
+                assert self.trunc_size == 0, \
+                    """To enable accumulated gradients,
+                       you must disable target sequence truncating."""
 
         # Set model in training mode.
         self.model.train()
 
+    def _accum_count(self, step):
+        for i in range(len(self.accum_steps)):
```

Comment:
here instead of looping every time shouldn't we update the `accum_count_l` list at each change so that all we need to check is `step > accum_count_l[0]`?